test/Feature/HLSLLib/select.int16.test

```diff
@@ -0,0 +1,166 @@
+#--- source.hlsl
+StructuredBuffer<bool> Cond : register(t0);
+StructuredBuffer<int16_t4> TrueVal0 : register(t1);
+StructuredBuffer<int16_t4> FalseVal0 : register(t2);
+StructuredBuffer<uint16_t4> TrueVal1 : register(t3);
+StructuredBuffer<uint16_t4> FalseVal1 : register(t4);
+
+RWStructuredBuffer<int16_t4> Out0 : register(u5);
+RWStructuredBuffer<uint16_t4> Out1 : register(u6);
+
+
+[numthreads(1,1,1)]
+void main() {
+  bool4 Cond0 = bool4(Cond[0], Cond[1], Cond[2], Cond[3]);
+  bool3 Cond1 = bool3(Cond[4], Cond[5], Cond[6]);
+  bool2 Cond2 = bool2(Cond[8], Cond[9]);
+  bool2 Cond3 = bool2(Cond[10], Cond[11]);
+
+  // int16_t
+  // vec4
+  Out0[0] = select(Cond0, TrueVal0[0], FalseVal0[0]);
+  Out0[1] = select(Cond0, TrueVal0[0].x, FalseVal0[0]);
+  Out0[2] = select(Cond0, TrueVal0[0], FalseVal0[0].x);
+  Out0[3] = select(Cond0, TrueVal0[0].x, FalseVal0[0].x);
+  // vec3 + scalar
+  Out0[4] = int16_t4(select(Cond1, TrueVal0[1].xyz, FalseVal0[1].xyz), select(Cond[7], TrueVal0[1].w, FalseVal0[1].w));
+  Out0[5] = int16_t4(select(Cond1, TrueVal0[1].x, FalseVal0[1].xyz), select(bool(1), int16_t(1), int16_t(-1)));
+  Out0[6] = int16_t4(select(Cond1, TrueVal0[1].xyz, FalseVal0[1].x), select(bool(0), int16_t(2), int16_t(-2)));
+  Out0[7] = int16_t4(select(Cond1, TrueVal0[1].x, FalseVal0[1].x), select(bool(1), int16_t(3), int16_t(-3)));
+  // vec2
+  Out0[8] = int16_t4(select(Cond2, TrueVal0[2].xy, FalseVal0[2].xy), select(Cond3, TrueVal0[2].z, FalseVal0[2].zw));

```

Comment:
Was the second select call intended to have TrueVal[2].z or should they both be TrueVal[2].zw?